airgun/entities/cloud_insights.py

```diff
@@ -0,0 +1,91 @@
+from wait_for import wait_for
+
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep
+from airgun.navigation import navigator
+from airgun.views.cloud_insights import CloudInsightsView
+from airgun.views.cloud_insights import CloudTokenView
+from airgun.views.job_invocation import JobInvocationCreateView
+
+
+class CloudInsightsEntity(BaseEntity):
+    endpoint_path = '/foreman_rh_cloud/insights_cloud'
+
+    def search(self, value):
+        """Search for 'query' and return matched hostnames/recommendations.
+
+        :param value: text to filter (default: no filter)
+        """
+        view = self.navigate_to(self, 'All')
+        return view.search(value)
+
+    def remediate(self, entity_name):
+        """Remediate hosts based on search input."""
+        view = self.navigate_to(self, 'All')
+        view.search(entity_name)
+        view.select_all.fill(True)
+        # view.select_all_hits.click() skip till BZ#1975321 is fixed.
+        view.remediate.click()
+        view.remediation_window.remediate.click()
+        self.run_job()
+
+    def sync_hits(self):
+        """Sync RH Cloud - Insights recommendations."""
+        view = self.navigate_to(self, 'All')
+        view.start_hits_sync.click()
+        self.browser.plugin.ensure_page_safe(timeout='60s')
+
+    def read(self, widget_names=None):
+        """Read all values for created activation key entity"""
+        view = self.navigate_to(self, 'All')
+        return view.read(widget_names=widget_names)
+
+    def save_token_sync_hits(self, value):
+        """Update Insights cloud view."""
+        view = self.navigate_to(self, 'Token')
+        view.rhcloud_token.fill(value)
+        view.save_token.click()
+        self.browser.plugin.ensure_page_safe(timeout='60s')
+
+    def update(self, values):
+        """Update RH Cloud - Insights view."""
+        view = self.navigate_to(self, 'All')
+        view.fill(values)
+
+    def run_job(self):
+        """Run remediation job"""
+        view = self.navigate_to(self, 'Run')
+        wait_for(
+            lambda: view.is_displayed is True,
```

Comment:
Same comment here as below - is there a reason `view.wait_displayed` can't be used here?

Reply:
updated the pr.